pkg/webui/components/location-form/index.js

```diff
@@ -0,0 +1,189 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { Component } from 'react'
+import { defineMessages } from 'react-intl'
+import * as Yup from 'yup'
+import bind from 'autobind-decorator'
+
+import Form from '../form'
+import Message from '../../lib/components/message'
+import Field from '../field'
+import SubmitBar from '../submit-bar'
+import Notification from '../notification'
+import ModalButton from '../button/modal-button'
+import Button from '../button'
+import toast from '../toast'
+
+import sharedMessages from '../../lib/shared-messages'
+import PropTypes from '../../lib/prop-types'
+
+import {
+  latitude as latitudeRegexp,
+  longitude as longitudeRegexp,
+  int32 as int32Regexp,
+} from '../../console/lib/regexp'
+
+const m = defineMessages({
+  deleteWarning: 'Are you sure you want to delete this location entry?',
+  deleteLocation: 'Remove location entry',
+  noLocationSet: 'There is currently no location information set',
+  updateSuccess: 'The location has been updated successfully',
+  deleteFailure: 'There was a problem removing the location',
+  deleteSuccess: 'The location has been removed successfully',
+})
+
+const validationSchema = Yup.object().shape({
+  latitude: Yup.string().required()
+    .matches(latitudeRegexp, sharedMessages.validateLatLong),
+  longitude: Yup.string().required()
+    .matches(longitudeRegexp, sharedMessages.validateLatLong),
+  altitude: Yup.string().matches(int32Regexp, sharedMessages.validateInt32).required(),
+})
+
+const defaultValues = {
+  longitude: undefined,
+  latitude: undefined,
+  altitude: undefined,
+}
+
+@bind
+class LocationForm extends Component {
+  constructor (props) {
+    super(props)
+
+    this.form = React.createRef()
+  }
+
+  state = {
+    error: '',
+  }
+
+  async onSubmit (values, { resetForm }) {
+    const { onSubmit, entityId } = this.props
+
+    this.setState({ error: '' })
+
+    try {
+      await onSubmit(values)
+      toast({
+        title: entityId,
+        message: m.updateSuccess,
+        type: toast.types.SUCCESS,
+      })
+    } catch (error) {
+      await this.setState({ error })
+    }
+
+    resetForm(values)
+  }
+
+  async onDelete () {
+    const { onDelete, entityId } = this.props
+
+    try {
+      await onDelete()
+      toast({
+        title: entityId,
+        message: m.deleteSuccess,
+        type: toast.types.SUCCESS,
+      })
+    } catch (error) {
+      toast({
+        title: entityId,
+        message: m.deleteFailure,
+        type: toast.types.ERROR,
+      })
+    }
+
+    this.form.current.resetForm()
```

Comment:
we have the `handleReset` prop in the `<Form />` component. is there a way to use this instead of the ref?

Reply:
I don't think so, I cannot `connect` into the formik context provider here, as it is the same component level. Using refs, in this case, was also how I saw it recommended on stack overflow.
If you have any other idea, let me know. Albeit, it works like this.